File: api/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;
use App\Models\Invoice;

class Order extends Model
{
    use SoftDeletes;
    protected $fillable = [
        'order_number', 'order_date', 'order_status_id', 'shipping_agency_id', 'purchase_type_id',
        'warehouse_id',
        'observations', 'phone', 'customer_id', 'customer_name', 'dni', 'province_id', 'district_id', 'address',
        'delivery_cost', 'total', 'document_type_id', 'document_number', 'customer_email', 'needs_receipt', 'user_id',
        'guide_transfer_reason_code', 'guide_transfer_reason_description', 'guide_transfer_mode', 'guide_transfer_date',
        'guide_total_weight', 'guide_weight_unit', 'guide_package_count',
        'guide_origin_ubigeo', 'guide_origin_address', 'guide_destination_ubigeo', 'guide_destination_address',
        'guide_recipient_doc_type', 'guide_recipient_doc_number', 'guide_recipient_name',
        'guide_carrier_doc_type', 'guide_carrier_doc_number', 'guide_carrier_name',
        'guide_vehicle_plate', 'guide_driver_doc_type', 'guide_driver_doc_number', 'guide_driver_name',
        'guide_driver_license', 'guide_transport_certificate',
        'guide_series', 'guide_correlativo', 'guide_full_number', 'guide_status', 'guide_sunat_code',
        'guide_sunat_description', 'guide_xml_content', 'guide_cdr_content', 'guide_sent_at',
    ];

    protected $casts = [
        'order_date' => 'datetime',
        'total' => 'decimal:2',
        'delivery_cost' => 'decimal:2',
        'needs_receipt' => 'boolean',
        'guide_transfer_date' => 'date',
        'guide_total_weight' => 'decimal:3',
        'guide_package_count' => 'integer',
        'guide_correlativo' => 'integer',
        'guide_sunat_code' => 'integer',
        'guide_sent_at' => 'datetime',
    ];

    public function orderStatus(): BelongsTo
    {
        return $this->belongsTo(OrderStatus::class, 'order_status_id');
    }

    public function shippingAgency(): BelongsTo
    {
        return $this->belongsTo(ShippingAgency::class, 'shipping_agency_id');
    }

    public function purchaseType(): BelongsTo
    {
        return $this->belongsTo(PurchaseType::class, 'purchase_type_id');
    }

    public function warehouse(): BelongsTo
    {
        return $this->belongsTo(Warehouse::class, 'warehouse_id');
    }

    public function customer(): BelongsTo
    {
        return $this->belongsTo(Customer::class, 'customer_id');
    }

    public function province(): BelongsTo
    {
        return $this->belongsTo(Province::class, 'province_id');
    }

    public function district(): BelongsTo
    {
        return $this->belongsTo(District::class, 'district_id');
    }

    public function documentType(): BelongsTo
    {
        return $this->belongsTo(DocumentType::class, 'document_type_id');
    }

    public function invoices(): HasMany
    {
        return $this->hasMany(Invoice::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    public function items(): HasMany
    {
        return $this->hasMany(OrderItem::class)->orderBy('sort_order');
    }
}

File: api-dotnet/AdminHiitop.Api/Domain/Sales/Entities/Order.cs
using AdminHiitop.Api.Domain.Catalog.Entities;
using AdminHiitop.Api.Domain.Common;

namespace AdminHiitop.Api.Domain.Sales.Entities;

public sealed class Order : AuditableEntity
{
    public string OrderNumber { get; set; } = string.Empty;
    public DateTime OrderDate { get; set; }
    public int OrderStatusId { get; set; }
    public int? ShippingAgencyId { get; set; }
    public int? PurchaseTypeId { get; set; }
    public int? WarehouseId { get; set; }
    public string? Observations { get; set; }
    public string? Phone { get; set; }
    public int? CustomerId { get; set; }
    public string? CustomerName { get; set; }
    public string? Dni { get; set; }
    public int? ProvinceId { get; set; }
    public int? DistrictId { get; set; }
    public string? Address { get; set; }
    public string? PickupKey { get; set; }
    public string? TrackingNumber { get; set; }
    public decimal DeliveryCost { get; set; }
    public decimal Total { get; set; }
    public int? DocumentTypeId { get; set; }
    public int? DocumentPrintFormatId { get; set; }
    public string? DocumentNumber { get; set; }
    public string? CustomerEmail { get; set; }
    public bool NeedsReceipt { get; set; }
    public int? UserId { get; set; }
    public string? GuideTransferReasonCode { get; set; }
    public string? GuideTransferReasonDescription { get; set; }
    public string? GuideTransferMode { get; set; }
    public DateTime? GuideTransferDate { get; set; }
    public decimal? GuideTotalWeight { get; set; }
    public string? GuideWeightUnit { get; set; }
    public int? GuidePackageCount { get; set; }
    public string? GuideOriginUbigeo { get; set; }
    public string? GuideOriginAddress { get; set; }
    public string? GuideDestinationUbigeo { get; set; }
    public string? GuideDestinationAddress { get; set; }
    public string? GuideRecipientDocType { get; set; }
    public string? GuideRecipientDocNumber { get; set; }
    public string? GuideRecipientName { get; set; }
    public string? GuideCarrierDocType { get; set; }
    public string? GuideCarrierDocNumber { get; set; }
    public string? GuideCarrierName { get; set; }
    public string? GuideVehiclePlate { get; set; }
    public string? GuideDriverDocType { get; set; }
    public string? GuideDriverDocNumber { get; set; }
    public string? GuideDriverName { get; set; }
    public string? GuideDriverLicense { get; set; }
    public string? GuideTransportCertificate { get; set; }
    public string? GuideSeries { get; set; }
    public int? GuideCorrelativo { get; set; }
    public string? GuideFullNumber { get; set; }
    public string? GuideStatus { get; set; }
    public int? GuideSunatCode { get; set; }
    public string? GuideSunatDescription { get; set; }
    public string? GuideXmlContent { get; set; }
    public string? GuideCdrContent { get; set; }
    public DateTime? GuideSentAt { get; set; }

    public OrderStatus OrderStatus { get; set; } = null!;
    public ShippingAgency? ShippingAgency { get; set; }
    public PurchaseType? PurchaseType { get; set; }
    public Warehouse? Warehouse { get; set; }
    public Customer? Customer { get; set; }
    public Province? Province { get; set; }
    public District? District { get; set; }
    public DocumentType? DocumentType { get; set; }
    public DocumentPrintFormat? DocumentPrintFormat { get; set; }
    public User? User { get; set; }
    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
    public ICollection<Invoice> Invoices { get; set; } = new List<Invoice>();
}
